### PR TITLE
[polaris.shopify.com] Fix content issues and typos

### PR DESCRIPTION
**Pull request by tjonx**
### WHAT is this pull request doing?

- Fixes typos
- Rephrases some directives
- Fixes wrong images/duplicate images in some directives
- Adds missing `icon` to some markdown pages and adjusts existing ones to be more relevant

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by dominikwilkowski — APPROVED** [review]

File: polaris.shopify.com/content/design/icons/creating-icons.md [hunk 1]
```diff
@@ -222,13 +220,14 @@ Outlined icons are the default. Filled icons are reserved for navigation only.
     ![A calendar, an inbox and a shop icon with the corner radius highlighted. 3 pixels for the calendar, 2 pixels for the inbox and 1 pixel for the inbox.](/images/design/icons/icons-creating-outline-5@2x.png)
 
     Corner radius can go from 1 px to 3 px, depending on the object roundness. Joins must be rounded. Sharp corners are not allowed, except for intersections and cutouts.
+    Corner radius can go from 1 px to 3 px, depending on the object roundness. Joins must be rounded. Sharp corners are not allowed, except for intersections and cutouts.
```
Comment (kyledurand) [inline polaris.shopify.com/content/design/icons/creating-icons.md]: Was this line duplicated or is github tripping?

**Review by kyledurand — COMMENTED** [review]
Looks good but it seems there was one line that's been duplicated by accident

File: polaris.shopify.com/content/design/icons/creating-icons.md (hunk 1 above)
Comment (tjonx) [reply polaris.shopify.com/content/design/icons/creating-icons.md]: Ah, probably a weird merge when I rebased. Thanks!

**Review by kyledurand — APPROVED** [review]
👍

**Pull request merged**